src/change_stream.ts

```diff
@@ -107,94 +108,220 @@ export interface PipeOptions {
+   * @see https://docs.mongodb.com/manual/changeStreams/#startafter-for-change-streams
+   */
   startAfter?: ResumeToken;
   /** Will start the changeStream after the specified operationTime. */
   startAtOperationTime?: OperationTime;
-  /** The number of documents to return per batch. See {@link https://docs.mongodb.com/manual/reference/command/aggregate|aggregation documentation}. */
+  /**
+   * The number of documents to return per batch.
+   * @see https://docs.mongodb.com/manual/reference/command/aggregate
+   */
   batchSize?: number;
 }
 
 /** @public */
-export interface ChangeStreamDocument<TSchema extends Document = Document> {
+export interface ChangeStreamFullNameSpace {
+  /** The namespace */
+  ns: { db: string; coll: string };
+}
+
+/** @public */
+export interface ChangeStreamDocumentKey<TSchema extends Document = Document> {
   /**
-   * The id functions as an opaque token for use when resuming an interrupted
-   * change stream.
+   * For unsharded collections this contains a single field `_id`.
+   * For sharded collections, this will contain all the components of the shard key
    */
-  _id: InferIdType<TSchema>;
+  documentKey: { _id: InferIdType<TSchema>; [shardKey: string]: any };
+}
 
+/** @public */
+export interface ChangeStreamDocumentCommon {
   /**
-   * Describes the type of operation represented in this change notification.
+   * The id functions as an opaque token for use when resuming an interrupted
+   * change stream.
    */
-  operationType:
-    | 'insert'
-    | 'update'
-    | 'replace'
-    | 'delete'
-    | 'invalidate'
-    | 'drop'
-    | 'dropDatabase'
-    | 'rename';
-
+  _id: ResumeToken;
   /**
-   * Contains two fields: “db” and “coll” containing the database and
-   * collection name in which the change happened.
+   * The timestamp from the oplog entry associated with the event.
+   * For events that happened as part of a multi-document transaction, the associated change stream
+   * notifications will have the same clusterTime value, namely the time when the transaction was committed.
+   * On a sharded cluster, events that occur on different shards can have the same clusterTime but be
+   * associated with different transactions or even not be associated with any transaction.
+   * To identify events for a single transaction, you can use the combination of lsid and txnNumber in the change stream event document.
    */
-  ns: { db: string; coll: string };
+  clusterTime?: Timestamp;
 
   /**
-   * Only present for ops of type ‘insert’, ‘update’, ‘replace’, and
-   * ‘delete’.
-   *
-   * For unsharded collections this contains a single field, _id, with the
-   * value of the _id of the document updated.  For sharded collections,
-   * this will contain all the components of the shard key in order,
-   * followed by the _id if the _id isn’t part of the shard key.
+   * The transaction number.
+   * Only present if the operation is part of a multi-document transaction.
```

Comment:
Can we note here that the type of transaction number depends on the `promoteLongs` flag?

Reply:
I'm wondering if it's more friction that it's worth to include this in the type annotation and instead document it as a possiblity based on the option.

Reply:
So include this as part of the documentation for the `promoteLongs` flag?  I'm okay with that